docker/start.sh

```diff
@@ -7,6 +7,8 @@ if [ -z "$DATABASE_URL" ]; then
   exit 1
 fi
 
+rails assets:precompile
```

Comment:
Unless I'm misunderstanding, it looks like this compiles assets when the app boots as opposed to when the image is built. That might be a bit of a performance annoyance if we were at Mastodon scale, but we've got a little ways to go, so I think this will do for now.

One other possibility would be to remove the `Dotenv` requirement, but that would make it less clear when someone is missing an essential key.

Reply:
> it looks like this compiles assets when the app boots as opposed to when the image is built.

Yeah, that's correct.

I'm not sure what the consequences are of removing the `Dotenv` requirement, but to make the precompilation work in the Dockerfile I also had to use default values in `config/applications.rb`, which (if we did both) would mean that someone could theoretically run the application unsafely with the defaults for their keys. Personally I'd rather play it safe than sorry and make those env vars a hard requirement.

Reply:
Yeah, or we could change the `.fetch` to instead be regular hash access, which won't raise an error:

```diff
-ENV.fetch('my_var')
+ENV['my_var']
```

But I agree, the hard requirement is probably better. Let's run with this for now.